azure/services/virtualmachines/virtualmachines.go

```diff
@@ -111,6 +115,34 @@ func (s *Service) Reconcile(ctx context.Context) error {
 		}
 		s.Scope.SetAddresses(addresses)
 		s.Scope.SetVMState(infraVM.State)
+
+		spec, ok := vmSpec.(*VMSpec)
+		if !ok {
+			return errors.New("invalid VM spec")
+		}
+		idsClient := identities.NewClient(s.Scope)
+
+		// Check if any userAssignedIdentities are missing from the VM and set a condition if so
+		if len(spec.UserAssignedIdentities) == len(infraVM.UserAssignedIdentities) {
+			for _, expectedIdentity := range spec.UserAssignedIdentities {
+				expectedClientID, err := idsClient.GetClientID(ctx, expectedIdentity.ProviderID)
+				if err != nil {
+					return errors.Wrap(err, "failed to get client ID")
+				}
+				found := false
+				for _, actualIdentity := range infraVM.UserAssignedIdentities {
```

Comment:
I'm actually not sure about this one. I had to use a nested for loop since we can't assume the userAssignedIdentities would be in the same order between the vm and the machine. I think I may be able to optimize this further using a map. cc @CecileRobertMichon on whether this might be too inefficient.

Reply:
I wasn't able to find any public docs about how many identities you can have assigned to a single VM but I agree that we don't expect more than a few per VM